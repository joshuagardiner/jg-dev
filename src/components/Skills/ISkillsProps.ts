import { ISkills } from "../../lib/models/ISkills";

/**
 * The properties available to the Skills component.
 */
export interface ISkillsProps {
  /**
   * The skills content.
   */
  skills: ISkills;
}
