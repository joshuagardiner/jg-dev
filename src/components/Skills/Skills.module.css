.skills {
  list-style: none;
  margin-top: -1.75%;
}
