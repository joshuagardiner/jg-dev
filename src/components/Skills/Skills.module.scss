.skill,
.skills {
  display: inline-block;
  font-size: 1.125rem;
  list-style: none;
  padding: 0 1.5vw 0 1.5vw;
}

.title {
  font-size: 2rem;
  margin-top: 7.5%;
}

@media screen and (max-width: 425px) {
  .skill {
    font-size: 0.75rem;
  }

  .title {
    font-size: 1.25rem;
  }
}
