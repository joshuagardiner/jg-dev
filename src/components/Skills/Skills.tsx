import React from "react";
import { Text, Title } from "@joshuagardiner/typescript-component-library";
import styling from "./Skills.module.css";

export const Skills: React.FC = () => {
  const skillsArray = [
    "React",
    "TypeScript",
    "CI/CD",
    "AWS",
    "Azure",
    "TDD",
    "GitHub",
    "Jest",
    "Cypress",
    "Terraform",
    "Docker",
    "C#",
    "Golang",
  ];

  const softSkillsArray = [
    "Communication",
    "Iniative",
    "Leadership",
    "Problem solving",
    "Teamwork",
  ];

  return (
    <>
      <Title fontSize={"2vw"} marginTop={"7.5%"} text={"Skills"} />
      <ul className={styling.skills}>
        {skillsArray.map((skill) => (
          <li key={skill}>{<Text fontSize={"1.125vw"} text={skill} />}</li>
        ))}
        {softSkillsArray.map((softSkill) => (
          <li key={softSkill}>
            {<Text fontSize={"1.125vw"} text={softSkill} />}
          </li>
        ))}
      </ul>
    </>
  );
};
