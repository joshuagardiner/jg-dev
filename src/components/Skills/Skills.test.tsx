import { render, screen } from "@testing-library/react";
import { Skills } from "./Skills";

describe("Skills", () => {
  it("renders the Skills component correctly.", () => {
    render(<Skills />);

    const skillsTitle = screen.getByTestId("title");
    const skillsList = screen.getByTestId("skills-list");
    const technicalSkills = screen.getAllByTestId("technical-skill");
    const softSkills = screen.getAllByTestId("soft-skill");

    expect(skillsTitle).toBeDefined();
    expect(skillsTitle.textContent).toEqual("Skills");
    expect(skillsList).toBeDefined();
    expect(technicalSkills).toHaveLength(13);
    expect(softSkills).toHaveLength(5);
  });

  it("renders the Skills component unchanged.", () => {
    const { container } = render(<Skills />);
    expect(container).toMatchSnapshot();
  });
});
