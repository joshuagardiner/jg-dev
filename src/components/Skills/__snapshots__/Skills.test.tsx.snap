// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders the Skills component unchanged. 1`] = `
<div>
  <div
    data-testid="skills-component"
  >
    <div
      class="tcl-title"
      data-testid="title"
    >
      <h1
        style="font-size: 2vw; margin-top: 7.5%;"
      >
        Skills
      </h1>
    </div>
    <ul
      class="skills"
      data-testid="skills-list"
    >
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            React
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            TypeScript
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            CI/CD
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            AWS
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            Azure
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            TDD
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            GitHub
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            Jest
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            Cypress
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            Terraform
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            Docker
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            C#
          </p>
        </div>
      </li>
      <li
        data-testid="technical-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            Golang
          </p>
        </div>
      </li>
      <li
        data-testid="soft-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            Communication
          </p>
        </div>
      </li>
      <li
        data-testid="soft-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            Iniative
          </p>
        </div>
      </li>
      <li
        data-testid="soft-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            Leadership
          </p>
        </div>
      </li>
      <li
        data-testid="soft-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            Problem solving
          </p>
        </div>
      </li>
      <li
        data-testid="soft-skill"
      >
        <div
          class="tcl-text"
          data-testid="text"
        >
          <p
            style="font-size: 1.125vw; margin-left: auto; margin-right: auto;"
          >
            Teamwork
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
